terraform/dotnet/ec2/default/main.tf

```
@@ -252,7 +256,11 @@ resource "null_resource" "remote_service_setup" {
       dotnet build
       export CORECLR_ENABLE_PROFILING=1
       export CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
-      export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+      if [ "${var.cpu_architecture}" == "arm64" ]; then
+        export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-arm64/OpenTelemetry.AutoInstrumentation.Native.so
+      else
+        export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+      fi
```

nit: Alternatively, could have kept it in one line by doing the following
```suggestion
      export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-${var.cpu_architecture == "arm64" ? "arm64" : "x64"}/OpenTelemetry.AutoInstrumentation.Native.so
```